InventorySales/Plugin/Sales/OrderManagement/AppendReservationsAfterOrderPlacementPlugin.php

```diff
@@ -160,6 +181,27 @@ public function afterPlace(OrderManagementInterface $subject, OrderInterface $or
         ]);
 
         $this->placeReservationsForSalesEvent->execute($itemsToSell, $salesChannel, $salesEvent);
+
+        try {
+            $order = $proceed($order);
+        } catch (\Exception $e) {
+            //add compensation
+            foreach ($itemsToSell as $item) {
+                $item->setQuantity(-(float)$item->getQuantity());
```

Comment:
we no need to fix Quantity here.
Ideally \Magento\InventorySalesApi\Api\Data\ItemToSellInterface should be immutable interface